applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_base_potential_flow_element.cpp

```diff
@@ -13,108 +13,85 @@
 #include "compressible_potential_flow_application_variables.h"
 #include "incompressible_potential_flow_element.h"
 #include "compressible_potential_flow_element.h"
-#include "adjoint_potential_flow_element.h"
+#include "adjoint_base_potential_flow_element.h"
+#include "custom_utilities/potential_flow_utilities.h"
 
 namespace Kratos
 {
     template <class TPrimalElement>
-    Element::Pointer AdjointPotentialFlowElement<TPrimalElement>::Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const 
+    Element::IntegrationMethod AdjointBasePotentialFlowElement<TPrimalElement>::GetIntegrationMethod() const
     {
-        KRATOS_TRY
-          return Kratos::make_intrusive<AdjointPotentialFlowElement<TPrimalElement>>(NewId, GetGeometry().Create(ThisNodes), pProperties);
-        KRATOS_CATCH("");
-    }
-
-    template <class TPrimalElement>
-    Element::Pointer AdjointPotentialFlowElement<TPrimalElement>::Create(IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties) const 
-    {
-        KRATOS_TRY
-            return Kratos::make_intrusive<AdjointPotentialFlowElement<TPrimalElement>>(NewId, pGeom, pProperties);
-        KRATOS_CATCH("");
-    }
-
-    template <class TPrimalElement>
-    Element::Pointer AdjointPotentialFlowElement<TPrimalElement>::Clone(IndexType NewId, NodesArrayType const& ThisNodes) const 
-    {
-        KRATOS_TRY
-        return Element::Pointer(new AdjointPotentialFlowElement(NewId, GetGeometry().Create(ThisNodes), pGetProperties()));
-        KRATOS_CATCH("");
+        return mpPrimalElement->GetIntegrationMethod();
     }
 
     template <class TPrimalElement>
-    Element::IntegrationMethod AdjointPotentialFlowElement<TPrimalElement>::GetIntegrationMethod() const 
+    void AdjointBasePotentialFlowElement<TPrimalElement>::Initialize()
     {
-        return mpPrimalElement->GetIntegrationMethod();
-    }
-  
-    template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::Initialize() 
-    {   
         mpPrimalElement->Initialize();
     }
 
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo) 
+    void AdjointBasePotentialFlowElement<TPrimalElement>::InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo)
     {
         mpPrimalElement->Data() = this->Data();
         mpPrimalElement->Set(Flags(*this));
         mpPrimalElement->InitializeSolutionStep(rCurrentProcessInfo);
```

Comment:
why is this done in `InitializeSolutionStep`?

Reply:
I added this in my first adjoint PR , since the "mpPrimalElement" depends on some flags to be properly solved. Since this "mpPrimalElement" (that comes from the template)  is empty, here I copy flags/data needed. 

I do this in the InitialiazeSolutionStep because the flags/data are set in other processes in the ExecuteInitialize/ExecuteInitializeSolutionStep.